.github/workflows/sonarcloud.yml

```diff
@@ -117,3 +117,11 @@ jobs:
           sonar-scanner-$SONAR_SCANNER_VERSION-linux/bin/sonar-scanner \
                 -Dsonar.login=$SONAR_TOKEN \
                 -Dsonar.cfamily.threads=$(nproc)
+
+    - name: Clean build directory
+      working-directory: ./build
+      run: |
+        make clean
```

Comment:
(same below) I didn't test it, but I think `git clean -xdf` will actually remove the whole `build` directory (that's created inside local sources).
Also `make clean` keeps `cmake` artefacts and cache, so if necessary, `rm -rf ./build` is probably more complete.